great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2867,6 +2869,10 @@ def _spark_map_condition_index(
     if unexpected_condition is None:
         return None
 
+    result_format = metric_value_kwargs["result_format"]
+    if not result_format.get("unexpected_index_column_names"):
+        return None
+
```

Comment:
Also allows for fallback behavior (Will return `None` instead of giving an error)

Reply:
Same note

Reply:
Why return `None` instead of an empty list? Could we keep the return non-null?

Reply:
`None` would determine whether the particular metric is returned in the EVR at the end. But I think the same behavior is doable eitehr way

Reply:
This one is going to add some complexity downstream (I think) because it is going to be a list with an empty dictionary, which is a little tricker to check for emptiness than `None` is.